packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -55,20 +66,48 @@ impl NetworkInfo {
 #[derive(Debug, Clone, Default, Zeroize, ZeroizeOnDrop)]
 struct UnresolvedSignatures {
     #[zeroize(skip)]
-    addr_idx_offset_map: HashMap<Bech32Address, u8>,
+    addr_idx_offset_map: HashMap<Bech32Address, u64>,
     secret_keys: Vec<SecretKey>,
 }
 
-pub trait TransactionBuilder: Send {
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+pub trait BuildableTransaction {
+    type TxType: Transaction;
+    /// Build a `Transaction` from the `TransactionBuilder`. `DryRunner` is
+    /// used to return the actual `gas_used` which is set as the `script_gas_limit`.
+    /// `CreateTransaction`s do not have `gas_limit` so the `DryRunner` is not used
```

Comment:
The part of the comment that references the `CreateTransaction` should be moved there.